sql/ddl.mdx

```diff
@@ -0,0 +1,48 @@
+---
+title: DDL Statements
+description: 'creating tables and views in Arroyo SQL'
+---
+In addition to its SELECT capabilities Arroyo supports several Data Description Language (DDL) capabilities.
+
+## CREATE TABLE
+Arroyo's CREATE TABLE statements come in three flavors: Create Table As, Memory Table and Connection Table.
+### CREATE TABLE AS
+This comamnd creates a table from the query included in it, e.g.
+```sql
+CREATE TABLE orders AS SELECT customer_id, order_id FROM orders;
+```
+The schema is inferred from the query and other queries within the same context can select from the new table.
+### CREATE TABLE (In-Memory)
+CREATE TABLE statements without any connection info are presumed to be in memory.
+It can be written to within the same query context and then read from. 
+Currently you can only insert into a given memory table once, but multiple sources can then select from it.
+For example, you could create an `orders` with a statement like
+```sql
+CREATE TABLE orders (customer_id INT, order_id INT);
+```
+At present, `CREATE VIEW` is simply an alias for creating a memory table.
+### CREATE TABLE (Connection)
+If you want to read or write using a pre-defined connection 
```

Comment:
This maybe needs some more explanation about how sources work within SQL.

Here's a stab at it:

--- 
Connection tables allow Arroyo to read and write to external systems like Kafka clusters. Connection tables can be created either in the Source and Sink tabs of the Web UI, or directly within SQL via a `CREATE TABLE` call.  

To create a connection table in SQL, you must have already configured the the connection information in the Connections tab so that Arroyo knows how to connect and authenticate against the external system. Each configured connection has a name, which is then referenced within the `CREATE TABLE` call. The general from is

```sql
CREATE TABLE <table name> (
  [<field name> <field type>,]..
) WITH (
  connection = '<connection name>',
  [connection options]
)
```

For example, to create a table that can read and write from a Kafka connection called `my_kafka` and a Kafka topic `order_top`:

```sql
CREATE TABLE orders (
  customer_id INT,
  order_id INT)
 WITH (
  connection = 'my_kafka',
  topic = 'order_topic',
  format = 'json'
);
```

---

We also need docs on each source / sink type that documents the available options, but I think that should be outside of the SQL docs. Once that exists we can link to it here.